experimental/BasisLieHighestWeight/src/WeylPolytope.jl

```diff
@@ -53,38 +53,98 @@ function convert_lattice_points_to_monomials(
+  L::LieAlgebraStructure,
+  weight::Vector{ZZRingElem},
+  zero_coordinates::Set{Any},
 )
   # calculate all integer solutions to the following linear program:
   # [   |              |    ]       [   x   ]      
-  # [weights[1]...weights[k]]   *   [   |   ]   =   weight 
+  # [weight_roots[1]...weight_roots[k]]   *   [   |   ]   =   weight
   # [   |              |    ]       [  res  ] 
   # [   |              |    ]       [   |   ]
   # where res[i] >= 0 for all i
-
-  n = length(weights)
+  wght = w_to_alpha(L, weight)
+  n = length(weight_roots)
   m = length(wght)
-  A = zero_matrix(QQ, 2m + n, n)
-  b = [zero(QQ) for _ in 1:(2m + n)]
+
+  #zero_coordinates = 0
+  #zero_coordinates = compute_zero_coordinates(weight_roots, L, weight, highest_weight) 
+  A = zero_matrix(QQ, 2m + n + length(zero_coordinates), n)
+  b = [zero(QQ) for _ in 1:(2m + n + length(zero_coordinates))]
+
   # equalities
   for i in 1:n
-    w = matrix(QQ, m, 1, weights[i])
+    w = matrix(QQ, m, 1, weight_roots[i])
     A[1:m, i] = w
     A[(m + 1):(2m), i] = -w
   end
+
   b[1:m] = wght
   b[(m + 1):(2m)] = -wght
   # non-negativity
   for i in 1:n
     A[2m + i, i] = -1
   end
+  for (j, i) in enumerate(zero_coordinates)
+    A[2m + n + j, i] = 1
+    b[2m + n + j] = 0
+  end
+  sol = Vector{ZZRingElem}.(lattice_points(polyhedron(A, b)))
+  return sol
+end
+
+@doc raw"""
+    compute_zero_coordinates(weight_roots::Vector{Vector{QQFieldElem}}, L::LieAlgebraStructure, weight::Vector{ZZRingElem}, highest_weight::Vector{ZZRingElem})
+
+Given a birational sequence, the action reads from right to left. If for the given highest_weight, the right most root vector acts by zero because it 
+is in the annihilator of the highest_weight vector, then we do not need to consider this coordinate at all when determining the possible monomials.
+This extends then further to the left to all coordinates  the annihilator is then the annihilator of the sudmodule generated by all root vectors at the right end of the birational sequence.
+"""
+function compute_zero_coordinates(
+  weight_roots::Vector{Vector{QQFieldElem}},
+  L::LieAlgebraStructure,
+  weight::Vector{ZZRingElem},
+  highest_weight::Vector{ZZRingElem},
+)
+  n = length(weight_roots)
+  m = length(weight)
+  a = cartan_matrix(L)
+  non_zeros = Set()
+  all = Set(1:m)
+  non_zeros = Set(findall(!iszero, highest_weight))
 
-  return Vector{ZZRingElem}.(lattice_points(polyhedron(A, b)))
+  zero_coordinates = Set()
+  c = n
+  check = false
+  collect = Set()
+  if c > 0 && !issubset(all, non_zeros)
+    for i in 1:m
+      if weight_roots[c][i] != 0 && i in non_zeros
+        check = true
+        push!(collect, i)
+      end
+    end
+    for i in collect
+      for j in 1:m
+        if a[i, j] != 0
+          push!(non_zeros, j)
+        end
+      end
+    end
+    if !check
+      push!(zero_coordinates, c)
+    end
+    c = c - 1
+    check = false
+  end
+  return zero_coordinates
```

Comment:
```suggestion
  zero_coordinates = Set{Int}()
  collect = Set{Int}()
  for c in n:-1:1
    length(non_zeros) == m && break
    check = false
    for i in non_zeros
      if weight_roots[c][i] != 0
        check = true
        push!(collect, i)
      end
    end
    union!(non_zeros, (j for j in 1:m if any(i -> a[i, j] != 0, collect)))
    if !check
      push!(zero_coordinates, c)
    end
  end
  return zero_coordinates
```
einige Vereinfachungen (Teil 2)